osu.Server.Spectator/Services/SharedInterop.cs

```diff
@@ -121,6 +130,24 @@ private async Task<string> runCommand(HttpMethod method, string command, dynamic
             }
         }
 
+        /// <summary>
+        /// Runs an interop command in a backgrounded scope.
+        /// </summary>
+        /// <param name="method">The HTTP method.</param>
+        /// <param name="command">The command to run.</param>
+        /// <param name="postObject">Any data to send.</param>
+        private void runCommandScoped(HttpMethod method, string command, dynamic? postObject = null)
+        {
+            _ = Task.Run(async () =>
+            {
+                using (var scope = scopeFactory.CreateScope())
+                {
+                    var scopedInterop = (SharedInterop)scope.ServiceProvider.GetRequiredService<ISharedInterop>();
+                    await scopedInterop.runCommand(method, command, postObject).ConfigureAwait(false);
+                }
+            });
```

Comment:
Two questions:

- I can understand fire-and-forget running this, but is this at least going to get logged if/when it fails? Does this even fail safely as written (i.e. won't it kill the entire service or something because unobserved)i?
- I have to wonder about playing the `IServiceScopeFactory` stuff - does `SharedInterop` even need to have a transient lifetime? I assume that was taken from new BSS, but there it doesn't really matter what lifetime it has, so I just picked transient as the most convenient / least likely to blow up in my face, but in here, it may actually be worth making it singleton lifetime and skip all of the scope games?

Reply:
So um... This PR kinda became a bit of a mess. Sorry about that - I can remake if requested.

Looks like making it a singleton is probably the more acceptable path, I've taken that route. In particular, note that we still can't use the `FireAndForget` helper and log the error because `MultiplayerHub.Error()` references `MultiplayerHub.Context` which throws on disposal (typical SignalR transitive hub stuff).

In any case, LIO errors are logged internally with a fix in https://github.com/ppy/osu-server-spectator/pull/276/commits/2972e369d637125aca5fc1727e77f528023bab92 (I considered this before...) I have tested that throwing an exception doesn't kill the server (unobserved generally don't anyway).